File: backend/app/constants/strategies.py
# backend/app/constants/strategies.py
"""
Константы для возможных значений полей стратегий.
Используются для валидации и в качестве справочника.
"""


class StrategyType:
    """Типы стратегий"""

    WARMUP = "warmup"
    POSITION_CHECK = "position_check"
    PROFILE_NURTURE = "profile_nurture"  # Новый тип - нагул профиля

    ALL = [WARMUP, POSITION_CHECK, PROFILE_NURTURE]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class DataSourceType:
    """Типы источников данных для стратегий"""

    MANUAL_LIST = "manual_list"  # Ручной ввод списка
    FILE_UPLOAD = "file_upload"  # Загрузка файла (.txt, .csv)
    URL_IMPORT = "url_import"  # Импорт по URL
    GOOGLE_SHEETS = "google_sheets"  # Импорт из Google Таблиц
    GOOGLE_DOCS = "google_docs"  # Импорт из Google Документов

    ALL = [MANUAL_LIST, FILE_UPLOAD, URL_IMPORT, GOOGLE_SHEETS, GOOGLE_DOCS]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class ExecutionStatus:
    """Статусы выполнения стратегий"""

    PENDING = "pending"  # Ожидает выполнения
    RUNNING = "running"  # Выполняется
    COMPLETED = "completed"  # Завершено успешно
    FAILED = "failed"  # Завершено с ошибкой

    ALL = [PENDING, RUNNING, COMPLETED, FAILED]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class WarmupType:
    """Типы стратегий прогрева"""

    DIRECT = "direct"  # Только прямые заходы на сайты
    SEARCH = "search"  # Только поиск в Яндексе
    MIXED = "mixed"  # Комбинированная стратегия

    ALL = [DIRECT, SEARCH, MIXED]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class ProfileNurtureType:
    """Типы стратегий нагула профиля"""

    SEARCH_BASED = "search_based"  # Через поиск в Яндексе
    DIRECT_VISITS = "direct_visits"  # Прямые заходы на сайты
    MIXED_NURTURE = "mixed_nurture"  # Комбинированный нагул

    ALL = [SEARCH_BASED, DIRECT_VISITS, MIXED_NURTURE]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class SearchEngineType:
    """Поисковые системы для нагула"""

    YANDEX_RU = "yandex.ru"
    YANDEX_BY = "yandex.by"
    YANDEX_KZ = "yandex.kz"
    YANDEX_TR = "yandex.tr"
    YANDEX_UA = "yandex.ua"
    MAIL_RU = "mail.ru"
    DZEN_RU = "dzen.ru"
    # Альтернативные домены
    YA_RU = "ya.ru"

    ALL = [
        YANDEX_RU,
        YANDEX_BY,
        YANDEX_KZ,
        YANDEX_TR,
        YANDEX_UA,
        MAIL_RU,
        DZEN_RU,
        YA_RU,
    ]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL

    @classmethod
    def get_yandex_domains(cls) -> list:
        """Получить только домены Яндекса"""
        return [
            cls.YANDEX_RU,
            cls.YANDEX_BY,
            cls.YANDEX_KZ,
            cls.YANDEX_TR,
            cls.YANDEX_UA,
            cls.YA_RU,
        ]


class QuerySourceType:
    """Типы источников запросов для нагула"""

    MANUAL_INPUT = "manual_input"  # Ручной ввод в textarea
    FILE_UPLOAD = "file_upload"  # Загрузка файла
    URL_ENDPOINT = "url_endpoint"  # URL для получения запросов
    GOOGLE_DOCS = "google_docs"  # Google документ
    GOOGLE_SHEETS = "google_sheets"  # Google таблица

    ALL = [MANUAL_INPUT, FILE_UPLOAD, URL_ENDPOINT, GOOGLE_DOCS, GOOGLE_SHEETS]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class CheckFrequency:
    """Частота проверки позиций"""

    DAILY = "daily"  # Ежедневно
    WEEKLY = "weekly"  # Еженедельно
    MONTHLY = "monthly"  # Ежемесячно
    CUSTOM = "custom"  # Пользовательское расписание (cron)

    ALL = [DAILY, WEEKLY, MONTHLY, CUSTOM]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


class DeviceType:
    """Типы устройств"""

    DESKTOP = "desktop"  # Десктоп
    MOBILE = "mobile"  # Мобильное устройство

    ALL = [DESKTOP, MOBILE]

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls.ALL


# Примеры конфигураций по умолчанию
DEFAULT_WARMUP_CONFIG = {
    "type": WarmupType.MIXED,
    "proportions": {"direct_visits": 30, "search_visits": 70},
    "min_sites": 3,
    "max_sites": 7,
    "session_timeout": 15,
    "yandex_domain": "yandex.ru",
    "device_type": DeviceType.DESKTOP,
}

DEFAULT_POSITION_CHECK_CONFIG = {
    "check_frequency": CheckFrequency.DAILY,
    "yandex_domain": "yandex.ru",
    "device_type": DeviceType.DESKTOP,
    "max_pages": 10,
    "custom_schedule": None,
    "behavior": {
        "random_delays": True,
        "scroll_pages": True,
        "human_like_clicks": True,
    },
}

DEFAULT_PROFILE_NURTURE_CONFIG = {
    "nurture_type": ProfileNurtureType.SEARCH_BASED,
    "target_cookies": {"min": 50, "max": 100},
    "session_config": {
        "timeout_per_site": 15,  # секунды
        "min_timeout": 10,
        "max_timeout": 30,
    },
    "search_engines": [SearchEngineType.YANDEX_RU],
    "queries_source": {
        "type": QuerySourceType.MANUAL_INPUT,
        "refresh_on_each_cycle": False,  # Для URL источников
    },
    "behavior": {
        "return_to_search": True,  # Возвращаться в поиск после сайта
        "close_browser_after_cycle": False,  # Закрывать браузер после цикла
        "emulate_human_actions": True,  # Эмулировать действия человека
        "scroll_pages": True,
        "random_clicks": True,
    },
    "proportions": {"search_visits": 70, "direct_visits": 30},  # Для смешанного типа
}


# Валидация конфигураций
def validate_warmup_config(config: dict) -> dict:
    """Валидация конфигурации прогрева"""
    validated = {**DEFAULT_WARMUP_CONFIG, **config}

    if not WarmupType.is_valid(validated.get("type")):
        raise ValueError(f"Invalid warmup type. Must be one of: {WarmupType.ALL}")

    proportions = validated.get("proportions", {})
    if validated["type"] == WarmupType.MIXED:
        if not proportions.get("direct_visits") or not proportions.get("search_visits"):
            raise ValueError(
                "Mixed strategy requires both direct_visits and search_visits proportions"
            )

    return validated


def validate_position_check_config(config: dict) -> dict:
    """Валидация конфигурации проверки позиций"""
    validated = {**DEFAULT_POSITION_CHECK_CONFIG, **config}

    if not CheckFrequency.is_valid(validated.get("check_frequency")):
        raise ValueError(
            f"Invalid check frequency. Must be one of: {CheckFrequency.ALL}"
        )

    if validated["check_frequency"] == CheckFrequency.CUSTOM and not validated.get(
        "custom_schedule"
    ):
        raise ValueError("Custom frequency requires custom_schedule (cron expression)")

    return validated


def validate_profile_nurture_config(config: dict) -> dict:
    """Валидация конфигурации нагула профиля"""
    validated = {**DEFAULT_PROFILE_NURTURE_CONFIG, **config}

    # Проверяем тип нагула
    if not ProfileNurtureType.is_valid(validated.get("nurture_type")):
        raise ValueError(
            f"Invalid nurture type. Must be one of: {ProfileNurtureType.ALL}"
        )

    # Проверяем целевое количество куков
    target_cookies = validated.get("target_cookies", {})
    min_cookies = target_cookies.get("min", 50)
    max_cookies = target_cookies.get("max", 100)

    if min_cookies <= 0 or max_cookies <= 0:
        raise ValueError("Target cookies must be positive numbers")

    if min_cookies > max_cookies:
        raise ValueError("Min cookies cannot be greater than max cookies")

    # Проверяем поисковые системы
    search_engines = validated.get("search_engines", [])
    if validated["nurture_type"] in [
        ProfileNurtureType.SEARCH_BASED,
        ProfileNurtureType.MIXED_NURTURE,
    ]:
        if not search_engines:
            raise ValueError("Search engines required for search-based nurturing")

        for engine in search_engines:
            if not SearchEngineType.is_valid(engine):
                raise ValueError(f"Invalid search engine: {engine}")

    # Проверяем источник запросов
    queries_source = validated.get("queries_source", {})
    source_type = queries_source.get("type")

    if validated["nurture_type"] in [
        ProfileNurtureType.SEARCH_BASED,
        ProfileNurtureType.MIXED_NURTURE,
    ]:
        if not QuerySourceType.is_valid(source_type):
            raise ValueError(
                f"Invalid query source type. Must be one of: {QuerySourceType.ALL}"
            )

    # Проверяем пропорции для смешанного типа
    if validated["nurture_type"] == ProfileNurtureType.MIXED_NURTURE:
        proportions = validated.get("proportions", {})
        if not proportions.get("search_visits") or not proportions.get("direct_visits"):
            raise ValueError(
                "Mixed nurture requires both search_visits and direct_visits proportions"
            )

    # Проверяем конфигурацию сессии
    session_config = validated.get("session_config", {})
    timeout = session_config.get("timeout_per_site", 15)
    min_timeout = session_config.get("min_timeout", 10)
    max_timeout = session_config.get("max_timeout", 30)

    if timeout <= 0 or min_timeout <= 0 or max_timeout <= 0:
        raise ValueError("Session timeouts must be positive numbers")

    if min_timeout > max_timeout:
        raise ValueError("Min timeout cannot be greater than max timeout")

    return validated

File: frontend/src/stores/strategies.ts
// frontend/src/stores/strategies.ts
import {defineStore} from 'pinia'
import {ref, computed} from 'vue'
import {api} from '@/api'

export interface Strategy {
    id: string
    user_id: string
    template_id?: string
    name: string
    strategy_type: 'warmup' | 'position_check' | 'profile_nurture'
    config: Record<string, any>
    created_at: string
    updated_at: string
    is_active: boolean
    data_sources: DataSource[]
}

export interface DataSource {
    id: string
    strategy_id: string
    source_type: 'manual_list' | 'file_upload' | 'url_import' | 'google_sheets' | 'google_docs'
    source_url?: string
    file_path?: string
    items_count: number
    is_active: boolean
    created_at: string
}

export interface StrategyTemplate {
    id: string
    name: string
    description?: string
    strategy_type: 'warmup' | 'position_check' | 'profile_nurture'
    config: Record<string, any>
    is_system: boolean
    created_at: string
}

export interface StrategyExecution {
    id: string
    strategy_id: string
    task_id?: string
    execution_type: 'warmup' | 'position_check' | 'profile_nurture'
    profile_id?: string
    parameters?: Record<string, any>
    result?: Record<string, any>
    status: 'pending' | 'running' | 'completed' | 'failed'
    started_at?: string
    completed_at?: string
    error_message?: string
}

export const useStrategiesStore = defineStore('strategies', () => {
    const strategies = ref<Strategy[]>([])
    const templates = ref<StrategyTemplate[]>([])
    const executions = ref<StrategyExecution[]>([])
    const loading = ref(false)
    const error = ref<string | null>(null)

    // Computed getters
    const warmupStrategies = computed(() =>
        strategies.value.filter(s => s.strategy_type === 'warmup')
    )

    const positionCheckStrategies = computed(() =>
        strategies.value.filter(s => s.strategy_type === 'position_check')
    )

    const warmupTemplates = computed(() =>
        templates.value.filter(t => t.strategy_type === 'warmup')
    )

    const positionCheckTemplates = computed(() =>
        templates.value.filter(t => t.strategy_type === 'position_check')
    )

    const profileNurtureStrategies = computed(() =>
        strategies.value.filter(s => s.strategy_type === 'profile_nurture')
    )

    const profileNurtureTemplates = computed(() =>
        templates.value.filter(t => t.strategy_type === 'profile_nurture')
    )

    // Default configurations
    function getDefaultWarmupConfig() {
        return {
            type: 'mixed',
            proportions: {
                direct_visits: 30,
                search_visits: 70
            },
            min_sites: 3,
            max_sites: 7,
            session_timeout: 15,
            yandex_domain: 'yandex.ru',
            device_type: 'desktop'
        }
    }

    function getDefaultPositionCheckConfig() {
        return {
            check_frequency: 'daily',
            yandex_domain: 'yandex.ru',
            device_type: 'desktop',
            max_pages: 10,
            custom_schedule: null,
            behavior: {
                random_delays: true,
                scroll_pages: true,
                human_like_clicks: true
            }
        }
    }

    function getDefaultProfileNurtureConfig() {
        return {
            nurture_type: 'search_based',
            target_cookies: {
                min: 50,
                max: 100
            },
            session_config: {
                timeout_per_site: 15,
                min_timeout: 10,
                max_timeout: 30
            },
            search_engines: ['yandex.ru'],
            queries_source: {
                type: 'manual_input',
                refresh_on_each_cycle: false
            },
            behavior: {
                return_to_search: true,
                close_browser_after_cycle: false,
                emulate_human_actions: true,
                scroll_pages: true,
                random_clicks: true
            },
            proportions: {
                search_visits: 70,
                direct_visits: 30
            }
        }
    }

    // API methods
    async function fetchStrategyTemplates(strategyType?: string) {
        try {
            loading.value = true
            error.value = null

            const params = strategyType ? {strategy_type: strategyType} : {}
            const response = await api.get('/strategies/templates', {params})
            templates.value = response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка загрузки шаблонов стратегий'
            throw err
        } finally {
            loading.value = false
        }
    }

    async function fetchStrategies(strategyType?: string) {
        try {
            loading.value = true
            error.value = null

            const params = strategyType ? {strategy_type: strategyType} : {}
            const response = await api.get('/strategies', {params})
            strategies.value = response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка загрузки стратегий'
            throw err
        } finally {
            loading.value = false
        }
    }

    async function createStrategy(strategyData: {
        template_id?: string
        name: string
        strategy_type: 'warmup' | 'position_check' | 'profile_nurture'
        config: Record<string, any>
    }) {
        try {
            loading.value = true
            error.value = null

            const response = await api.post('/strategies', strategyData)
            strategies.value.push(response.data)

            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка создания стратегии'
            throw err
        } finally {
            loading.value = false
        }
    }

    async function updateStrategy(strategyId: string, updateData: {
        name?: string
        config?: Record<string, any>
        is_active?: boolean
    }) {
        try {
            loading.value = true
            error.value = null

            const response = await api.put(`/strategies/${strategyId}`, updateData)

            // Обновляем стратегию в локальном состоянии
            const index = strategies.value.findIndex(s => s.id === strategyId)
            if (index !== -1) {
                strategies.value[index] = {...strategies.value[index], ...response.data}
            }

            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка обновления стратегии'
            throw err
        } finally {
            loading.value = false
        }
    }

    async function deleteStrategy(strategyId: string) {
        try {
            loading.value = true
            error.value = null

            await api.delete(`/strategies/${strategyId}`)

            // Удаляем стратегию из локального состояния
            const index = strategies.value.findIndex(s => s.id === strategyId)
            if (index !== -1) {
                strategies.value.splice(index, 1)
            }
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка удаления стратегии'
            throw err
        } finally {
            loading.value = false
        }
    }

    async function addDataSource(strategyId: string, sourceData: {
        source_type: string
        source_url?: string
        data_content?: string
    }) {
        try {
            const response = await api.post(`/strategies/${strategyId}/data-sources`, sourceData)

            // Обновляем стратегию в локальном состоянии
            const strategy = strategies.value.find(s => s.id === strategyId)
            if (strategy) {
                strategy.data_sources.push(response.data)
            }

            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка добавления источника данных'
            throw err
        }
    }

    async function uploadDataFile(strategyId: string, file: File) {
        try {
            const formData = new FormData()
            formData.append('file', file)

            const response = await api.post(
                `/strategies/${strategyId}/data-sources/upload-file`,
                formData,
                {
                    headers: {
                        'Content-Type': 'multipart/form-data'
                    }
                }
            )

            // Обновляем стратегию
            await fetchStrategies()

            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка загрузки файла'
            throw err
        }
    }

    async function executeStrategy(strategyId: string, executionParams: Record<string, any> = {}) {
        try {
            loading.value = true
            error.value = null

            const response = await api.post(`/strategies/${strategyId}/execute`, {
                execution_params: executionParams
            })

            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка запуска стратегии'
            throw err
        } finally {
            loading.value = false
        }
    }

    async function fetchStrategyExecutions(strategyId?: string) {
        try {
            const params = strategyId ? {strategy_id: strategyId} : {}
            const response = await api.get('/strategies/executions', {params})
            executions.value = response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка загрузки выполнений стратегий'
            throw err
        }
    }

    async function assignStrategiesToProject(assignment: {
        domain_id?: string
        warmup_strategy_id?: string
        position_check_strategy_id?: string
        profile_nurture_strategy_id?: string
    }) {
        try {
            const response = await api.post('/strategies/project-strategies', assignment)
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка назначения стратегий'
            throw err
        }
    }

    async function getProjectStrategies(domainId?: string) {
        try {
            const params = domainId ? {domain_id: domainId} : {}
            const response = await api.get('/strategies/project-strategies', {params})
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка получения назначенных стратегий'
            throw err
        }
    }

    // Profile Nurture specific methods
    async function getAvailableSearchEngines() {
        try {
            const response = await api.get('/strategies/profile-nurture/search-engines')
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка получения поисковых систем'
            throw err
        }
    }

    async function validateProfileNurtureConfig(config: Record<string, any>) {
        try {
            const response = await api.post('/strategies/profile-nurture/validate-config', config)
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка валидации конфигурации'
            throw err
        }
    }

    async function testQuerySource(strategyId: string, sourceData: {
        source_type: string
        source_url?: string
        data_content?: string
    }) {
        try {
            const response = await api.post(`/strategies/${strategyId}/test-query-source`, sourceData)
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка тестирования источника'
            throw err
        }
    }

    async function getNurtureProgress(strategyId: string) {
        try {
            const response = await api.get(`/strategies/${strategyId}/nurture-progress`)
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка получения прогресса нагула'
            throw err
        }
    }

    async function getDefaultConfigs() {
        try {
            const response = await api.get('/strategies/default-configs')
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка получения конфигураций по умолчанию'
            throw err
        }
    }

    // Helper methods for strategy type display
    function getStrategyTypeLabel(type: string): string {
        switch (type) {
            case 'warmup':
                return 'Прогрев'
            case 'position_check':
                return 'Проверка позиций'
            case 'profile_nurture':
                return 'Нагул профиля'
            default:
                return type
        }
    }

    function getStrategyTypeDescription(type: string): string {
        switch (type) {
            case 'warmup':
                return 'Прогрев сайтов перед проверкой позиций'
            case 'position_check':
                return 'Проверка позиций сайтов в поиске'
            case 'profile_nurture':
                return 'Нагул профилей браузера для обхода детекции'
            default:
                return ''
        }
    }

    function getStrategyTypeIcon(type: string): string {
        switch (type) {
            case 'warmup':
                return 'BeakerIcon'
            case 'position_check':
                return 'ChartBarIcon'
            case 'profile_nurture':
                return 'UserIcon'
            default:
                return 'CogIcon'
        }
    }

    function getNurtureTypeLabel(type: string): string {
        switch (type) {
            case 'search_based':
                return 'Через поиск'
            case 'direct_visits':
                return 'Прямые заходы'
            case 'mixed_nurture':
                return 'Смешанный'
            default:
                return type
        }
    }

    // Data import methods for profile nurture
    async function importDataFromUrl(url: string) {
        try {
            const response = await api.post('/strategies/import/url', {url})
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка импорта данных по URL'
            throw err
        }
    }

    async function importDataFromGoogleSheets(url: string) {
        try {
            const response = await api.post('/strategies/import/google-sheets', {url})
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка импорта данных из Google Таблиц'
            throw err
        }
    }

    async function importDataFromGoogleDocs(url: string) {
        try {
            const response = await api.post('/strategies/import/google-docs', {url})
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка импорта данных из Google Документов'
            throw err
        }
    }

    async function validateDataSource(sourceData: {
        source_type: string
        source_url?: string
        data_content?: string
    }) {
        try {
            const response = await api.post('/strategies/validate-data-source', sourceData)
            return response.data
        } catch (err: any) {
            error.value = err.response?.data?.detail || 'Ошибка валидации источника данных'
            throw err
        }
    }

    // File processing methods
    async function processTextFile(file: File) {
        try {
            const content = await readFileAsText(file)
            return {
                type: 'manual_input',
                data_content: content,
                items_count: content.split('\n').filter(line => line.trim()).length
            }
        } catch (err: any) {
            error.value = 'Ошибка чтения текстового файла'
            throw err
        }
    }

    async function processCsvFile(file: File) {
        try {
            const content = await readFileAsText(file)
            // Простая обработка CSV - берем первую колонку
            const lines = content.split('\n')
            const processedLines = lines.map(line => {
                const cells = line.split(',')
                return cells[0]?.trim() || ''
            }).filter(line => line)

            return {
                type: 'manual_input',
                data_content: processedLines.join('\n'),
                items_count: processedLines.length
            }
        } catch (err: any) {
            error.value = 'Ошибка чтения CSV файла'
            throw err
        }
    }

    function readFileAsText(file: File): Promise<string> {
        return new Promise((resolve, reject) => {
            const reader = new FileReader()
            reader.onload = (e) => {
                const content = e.target?.result as string
                resolve(content)
            }
            reader.onerror = reject
            reader.readAsText(file)
        })
    }

    function getDataSourceStats(source: any) {
        if (!source.data_content) return {count: 0, preview: []}

        const lines = source.data_content.split('\n').filter(line => line.trim())
        return {
            count: lines.length,
            preview: lines.slice(0, 5),
            isEmpty: lines.length === 0,
            isMinimal: lines.length < 10,
            isGood: lines.length >= 10
        }
    }

    function getEnhancedProfileNurtureConfig() {
        return {
            nurture_type: 'search_based',
            target_cookies: {
                min: 50,
                max: 100
            },
            session_config: {
                timeout_per_site: 15,
                min_timeout: 10,
                max_timeout: 30
            },
            search_engines: ['yandex.ru'],
            queries_source: {
                type: 'manual_input',
                data_content: '',
                source_url: '',
                refresh_on_each_cycle: false
            },
            behavior: {
                return_to_search: true,
                close_browser_after_cycle: false,
                emulate_human_actions: true,
                scroll_pages: true,
                random_clicks: true
            },
            proportions: {
                search_visits: 70,
                direct_visits: 30
            },
            direct_sites_source: {
                type: 'manual_input',
                data_content: '',
                source_url: '',
                refresh_on_each_cycle: false
            }
        }
    }

    function cleanProfileNurtureConfig(config) {
        const cleaned = {...config}

        // Удаляем поля, которые не нужны для определенных типов
        if (cleaned.nurture_type === 'search_based') {
            delete cleaned.direct_sites_source
            delete cleaned.proportions
        } else if (cleaned.nurture_type === 'direct_visits') {
            delete cleaned.search_engines
            delete cleaned.proportions
            // Очищаем queries_source, но оставляем пустую структуру
            cleaned.queries_source = {
                type: 'manual_input',
                data_content: '',
                source_url: '',
                refresh_on_each_cycle: false
            }
        }

        return cleaned
    }

    function validateQueriesSource(source) {
        const errors = []

        if (!source) {
            errors.push('Источник запросов обязателен')
            return errors
        }

        if (source.type === 'manual_input' && !source.data_content?.trim()) {
            errors.push('Поисковые запросы не могут быть пустыми')
        }

        if (['url_import', 'google_sheets', 'google_docs'].includes(source.type) && !source.source_url?.trim()) {
            errors.push('URL источника запросов обязателен')
        }

        return errors
    }

    function validateDirectSitesSource(source) {
        const errors = []

        if (!source) {
            errors.push('Источник сайтов обязателен')
            return errors
        }

        if (source.type === 'manual_input' && !source.data_content?.trim()) {
            errors.push('Список сайтов не может быть пустым')
        }

        if (['url_import', 'google_sheets', 'google_docs'].includes(source.type) && !source.source_url?.trim()) {
            errors.push('URL источника сайтов обязателен')
        }

        return errors
    }

    async function createTemporaryStrategy(strategyData) {
        try {
            const response = await api.post('/strategies/temporary', strategyData)
            return response.data
        } catch (err) {
            error.value = err.response?.data?.detail || 'Ошибка создания временной стратегии'
            throw err
        }
    }

    return {
        // State
        strategies,
        templates,
        executions,
        loading,
        error,

        // Computed
        warmupStrategies,
        positionCheckStrategies,
        profileNurtureStrategies,
        warmupTemplates,
        positionCheckTemplates,
        profileNurtureTemplates,

        // Methods
        fetchStrategyTemplates,
        fetchStrategies,
        createStrategy,
        updateStrategy,
        deleteStrategy,
        addDataSource,
        uploadDataFile,
        executeStrategy,
        fetchStrategyExecutions,
        assignStrategiesToProject,
        getProjectStrategies,

        // Profile Nurture specific
        getAvailableSearchEngines,
        validateProfileNurtureConfig,
        testQuerySource,
        getNurtureProgress,
        getDefaultConfigs,

        // Default configs
        getDefaultWarmupConfig,
        getDefaultPositionCheckConfig,
        getDefaultProfileNurtureConfig,

        // Helper methods
        getStrategyTypeLabel,
        getStrategyTypeDescription,
        getStrategyTypeIcon,
        getNurtureTypeLabel,

        importDataFromUrl,
        importDataFromGoogleSheets,
        importDataFromGoogleDocs,
        validateDataSource,
        createTemporaryStrategy,
        processTextFile,
        processCsvFile,
        readFileAsText,
        getEnhancedProfileNurtureConfig,
        validateQueriesSource,
        validateDirectSitesSource,
        getDataSourceStats,

        cleanProfileNurtureConfig
    }
})

